lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -1706,6 +1706,15 @@ def cmd_db_connect_help
 
   def cmd_db_connect(*args)
     return if not db_check_driver
+    if framework.db.connection_established?
+      cdb = ""
+      ::ActiveRecord::Base.connection_pool.with_connection { |conn|
+        if conn.respond_to? :current_database
+          cdb = conn.current_database
+        end
+      } 
+      return print_status("#{framework.db.driver} already connected to #{cdb}")
```

Comment:
Sweet, thanks @nstarke! One minor comment here: might want to follow this with something along the lines of "Run db_disconnect first if you wish to connect to a different DB."